renku/core/management/datasets.py

```diff
@@ -476,59 +476,72 @@ def is_protected_path(self, path):
 
         return False
 
-    def _add_from_local(self, dataset, path, external, destination):
+    def _add_from_local(self, dataset, path, external, destination) -> List[Dict]:
         """Add a file or directory from a local filesystem."""
-        src = Path(os.path.abspath(path))
+        action = "symlink" if external else "copy"
```

Comment:
Not something you need to change here, but `action` seems to be a good candidate for using an Enum